exotica_python/src/Exotica.cpp

```diff
@@ -572,49 +587,11 @@ PYBIND11_MODULE(_pyexotica, module)
     scene.def("loadSceneFile", &Scene::loadSceneFile);
     scene.def("getScene", &Scene::getScene);
     scene.def("cleanScene", &Scene::cleanScene);
-
-    // CollisionScene-related functions exposed via Scene - NB: may change in future
-    scene.def("isStateValid", [](Scene* instance) {
-        return instance->getCollisionScene()->isStateValid();
-    });
-    scene.def("isStateValid", [](Scene* instance, bool self, double dist) {
-        return instance->getCollisionScene()->isStateValid(self, dist);
-    });
-    scene.def("updateWorld", [](Scene* instance, moveit_msgs::PlanningSceneWorld& world) {
-        moveit_msgs::PlanningSceneWorldConstPtr myPtr(new moveit_msgs::PlanningSceneWorld(world));
-        instance->getCollisionScene()->updateWorld(myPtr);
-    });
-    scene.def("getCollisionRobotLinks", [](Scene* instance) {
-        return instance->getCollisionScene()->getCollisionRobotLinks();
-    });
-    scene.def("getCollisionWorldLinks", [](Scene* instance) {
-        return instance->getCollisionScene()->getCollisionWorldLinks();
-    });
-    scene.def("getRobotDistance", [](Scene* instance, std::string link, bool self, double safeDist) {
-        double d;
-        Eigen::Vector3d p1, p2, norm, c1, c2;
-        instance->getCollisionScene()->getRobotDistance(link, self, d, p1, p2, norm, c1, c2, safeDist);
-        return py::make_tuple(d, norm);
-    });
-    scene.def("getDistance", [](Scene* instance, std::string o1, std::string o2,
-                                double safeDist) {
-      double d;
-      instance->getCollisionScene()->getDistance(o1, o2, d, safeDist);
-      return d;
-    });
-    scene.def("getDistance", [](Scene* instance, std::string o1, std::string o2,
-                                double safeDist,
-                                bool computeContactInformation) {
-      double d;
-      Eigen::Vector3d p1 = Eigen::Vector3d::Zero(),
-                      p2 = Eigen::Vector3d::Zero();
-      if (computeContactInformation) {
-        instance->getCollisionScene()->getDistance(o1, o2, d, p1, p2, safeDist);
-      } else {
-        instance->getCollisionScene()->getDistance(o1, o2, d, safeDist);
-      }
-      return py::make_tuple(d, p1, p2);
-    });
+    scene.def("isStateValid", [](Scene* instance, bool self) {return instance->getCollisionScene()->isStateValid(self);}, py::arg("self")=true);
```

Comment:
I'd like the margin/safe distance back - we used it heavily in the past (both in C++ and in Python).

Reply:
See above.